data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -0,0 +1,213 @@
+            if (isRangeOverlapping(rangeEntry)) {
+                String exceptionMsg = "map option contains key "+mapEntry.getKey()+" that overlaps with other range entries";
+                throw new InvalidPluginConfigurationException(exceptionMsg);
+            } else {
+                rangeMappings.put(Range.between(lowKey, highKey), mapEntry.getValue());
+            }
+        }
+    }
+
+    private void addIndividualMapping(String key, String value){
+        if(individualMappings.containsKey(key)){
+            String exceptionMsg = "map option contains duplicate entries of "+key;
+            throw new InvalidPluginConfigurationException(exceptionMsg);
+        }
+        else{
+            individualMappings.put(key.strip(), value);
+        }
+    }
+
+    private boolean isRangeOverlapping(Range<Float> rangeEntry){
+        for(Range<Float> range : rangeMappings.keySet()){
+            if(range.isOverlappedBy(rangeEntry)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void checkOverlappingKeys(){
+        for(String individualKey : individualMappings.keySet()){
+            if(NumberUtils.isParsable(individualKey)){
+                Float floatKey = Float.parseFloat(individualKey);
+                Range<Float> range = Range.between(floatKey, floatKey);
+                if(isRangeOverlapping(range)){
+                    String exceptionMsg = "map option contains key "+individualKey+" that overlaps with other range entries";
+                    throw new InvalidPluginConfigurationException(exceptionMsg);
+                }
+            }
+        }
+    }
+
+    private void parseFile(String filePath){
+        //todo
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
+        //todo
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            if (Objects.nonNull(translateProcessorConfig.getMapWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getMapWhen(), recordEvent)) {
+                continue;
+            }
+            try {
+                String matchKey = record.getData().get(translateProcessorConfig.getSource(), String.class);
+                if(matchesIndividualEntry(record, matchKey) || matchesRangeEntry(record, matchKey) || matchesPatternEntry(record, matchKey)){
+                    continue;
+                }
+                else{
+
+                    // todo : add default, increment metrics, and/or add_tags
+
+                }
+            } catch (Exception ex){
+                LOG.error(EVENT, "Error mapping the source [{}] of entry [{}]",
+                        translateProcessorConfig.getSource(), record.getData(), ex);
+            }
+        }
+        return records;
+    }
+
+    public boolean matchesIndividualEntry(Record<Event> record, String matchKey){
```

Comment:
This function and the two other `matches` functions below, pass `Event`, instead of `Record<Event>` like
 ```
public boolean matchesIndividualEntry(Event event, String matchKey){
```